kube-client/src/lib.rs

```diff
@@ -611,4 +611,140 @@ mod test {
         csr.delete(csr_name, &DeleteParams::default()).await?;
         Ok(())
     }
+
+    k8s_openapi::k8s_if_ge_1_25! {
```

Comment:
This does not seem to work :thinking: , v1.21 integration tests are still failing due to this test.

Reply:
huh. that's odd. i'm not sure i fully grok [the macro](https://docs.rs/k8s-openapi/latest/src/k8s_openapi/opt/rustwide/target/x86_64-unknown-linux-gnu/debug/build/k8s-openapi-5fdad33d0cb69d91/out/conditional_compilation_macros.rs.html#86), but thought for sure this would not get run with that..

maybe you have to move it to the bottom in a top level `k8s_if_ge_1_25! { #[cfg(test)] mod test {...} }`. might be able to test this locally with whatever thing you have running locally by varying the version to match your local cluster (if you have one)

Reply:
The macro actually seems to be working. The issue is that it looks at the features enabled for `k8s-openapi` which has nothing to do with the API version of the cluster. [this](https://github.com/jmintb/kube-rs/blob/82512c732dcd80325ca3a7214f2fa8af62ba4f1c/kube-client/Cargo.toml#L88) line enables the `v1_26` feature during tests, which intern causes the macro to include the ephemeral containers test.

I am not sure what would be the best way to get around this, perhaps a check inside the test for the actual cluster version? seems a bit hacky though.

Reply:
Ah, damn. I guess we've never have tests that actually had tests that target features like this. In the past I think we've stuffed them in examples and only run them as a one-off for the version.

..you could possibly use [`Client::apiserver_version`](https://docs.rs/kube/latest/kube/struct.Client.html#method.apiserver_version), the result struct contains a [major and a minor](https://docs.rs/k8s-openapi/0.18.0/k8s_openapi/apimachinery/pkg/version/struct.Info.html). I think that's actually sensible.

Reply:
Awesome! I will give that shot.